agasc/supplement/utils.py

```diff
@@ -134,3 +144,191 @@ def save_version(filename, table_name):
                    append=True, overwrite=True)
     last_updated.write(str(filename), format='hdf5', path='last_updated',
                        append=True, overwrite=True)
+
+
+def update_table(filename, table, path, dtype, keys=['agasc_id'], dry_run=False, create=False):
+    """
+    Update a table of the AGASC supplement.
+
+    This overwrites all rows already in the supplement table and appends that rows not yet in the
+    supplement table.
+
+    :param filename:
+    :param table: table.Table
+    :param path: str
+        the path of the table in the supplement ('obs', 'mags')
+    :param dtype: np.dtype
+        the dtype of the table. Used only if the table is not in the supplement already.
+        If it is not given, and the table does not exist in the supplement,
+        an exception will be raised.
+    :param keys: list
+        a list of columns that uniquely identify a row in the table.
+        In case of duplicates, the first appearance is kept.
+    :param dry_run: bool
+    :param create: bool
+        Create a supplement file if it does not exist
+    """
+    filename = Path(filename)
+
+    if len(table) == 0:
+        logger.info('Nothing to update')
+        return
+
+    suppl_table = get_table(filename, path, dtype, create=create)
+    table = Table(table if len(table) else None, dtype=dtype)
+    table = unique(table, keys=keys, keep='first')
+
+    # for now I require no masked elements.
+    if table.mask is not None and np.any(table.mask):
+        masked = [name for name in table.colnames if np.any(table[name].mask)]
+        raise Exception('"{path}" table should have no masked elements, '
+                        'but element in these columns are masked: ',
+                        ', '.join(masked))
+
+    # entries already in supplement
+    intersect = (table[keys[0]][None] == suppl_table[keys[0]][:, None])
```

Comment:
To make this approach more obvious/readable  please use `table[keys[0]][None, :]`.

Reply:
Done.